### PR TITLE
fix(php): Correct uint32 type mapping to use UInt32 instead of Int32

### PR DESCRIPTION
**Pull request by TomA-R**
This fixes the PHP extension's mapping for `uint32`, which was incorrectly using `Int32`. I've also added a couple of small tests to cover edge cases in `RepeatedField` and `MapField`

### CODE REVIEW

File: php/ext/google/protobuf/tests/php_uint32_mapfield_coercion.phpt [hunk 1]
```diff
@@ -0,0 +1,43 @@
+--TEST--
```
Comment (honglooker) [inline php/ext/google/protobuf/tests/php_uint32_mapfield_coercion.phpt]: Mind adding a blurb on copyright? In cpp code, it's generally of the form:
```
// Protocol Buffers - Google's data interchange format
// Copyright 2025 Google LLC.  All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd
```

Please convert this to the equivalent in php tests - thank you

File: php/ext/google/protobuf/tests/php_uint32_repeatedfield_coercion.phpt [hunk 2]
```diff
@@ -0,0 +1,42 @@
+--TEST--
```
Comment (honglooker) [inline php/ext/google/protobuf/tests/php_uint32_repeatedfield_coercion.phpt]: same here

**Review by honglooker — COMMENTED** [review]
Thanks for your patch! Added commentary below.

**Comment by honglooker** [comment]
Alternatively, it may also make sense to augment php/tests/GeneratedClassTest.php instead.

**Comment by TomA-R** [comment]
> Alternatively, it may also make sense to augment php/tests/GeneratedClassTest.php instead.

Thank you for your review, @honglooker! I've added the tests to php/tests/GeneratedClassTest.php and removed the original PHPT tests. I hope that's what you expected, let me know if you'd like me to make any other changes

File: php/ext/google/protobuf/tests/php_uint32_mapfield_coercion.phpt (hunk 1 above)
Comment (TomA-R) [reply php/ext/google/protobuf/tests/php_uint32_mapfield_coercion.phpt]: I've removed this file in favour of test additions to php/tests/GeneratedClassTest.php

File: php/ext/google/protobuf/tests/php_uint32_repeatedfield_coercion.phpt (hunk 2 above)
Comment (TomA-R) [reply php/ext/google/protobuf/tests/php_uint32_repeatedfield_coercion.phpt]: I've removed this file in favour of test additions to php/tests/GeneratedClassTest.php

File: php/tests/GeneratedClassTest.php [hunk 3]
```diff
@@ -269,6 +269,43 @@ public function testUint32Field()
         $this->assertSame(MIN_INT32, $m->getOptionalUint32());
     }
 
+    public function testUint32ArrayConstructor()
+    {
+        $m1 = new TestMessage(['optional_uint32' => 2147483648]);
+        $this->assertSame(MIN_INT32, $m1->getOptionalUint32());
+
+        $m2 = new TestMessage(['optional_uint32' => MAX_UINT32]);
+        $this->assertSame(-1, $m2->getOptionalUint32());
```
Comment (honglooker) [inline php/tests/GeneratedClassTest.php:278]: Does this test pass if you change the lefthand side from `-1` to `MAX_UINT32`?

File: php/tests/GeneratedClassTest.php [hunk 4]
```diff
@@ -269,6 +269,43 @@ public function testUint32Field()
...
+    }
+
+    public function testRepeatedUint32Boundaries()
+    {
+        $m = new TestMessage();
+        $rf = $m->getRepeatedUint32();
+
+        $rf[] = 0;
+        $rf[] = MAX_INT32;
+        $rf[] = 2147483648;
```
Comment (honglooker) [inline php/tests/GeneratedClassTest.php:288]: prob ezier to grok if `MAX_INT32 + 1`

File: php/tests/GeneratedClassTest.php [hunk 5]
```diff
@@ -269,6 +269,43 @@ public function testUint32Field()
...
+    public function testMapUint32Uint32Boundaries()
+    {
+        $m = new TestMessage();
+        $map = $m->getMapUint32Uint32();
+
+        $map[1] = 0;
+        $map[MAX_UINT32] = MAX_UINT32;
+
+        $this->assertSame(0, $map[1]);
+        $this->assertSame(-1, $map[MAX_UINT32]);
```
Comment (honglooker) [inline php/tests/GeneratedClassTest.php:306]: MAX_UINT32 ?

File: php/tests/GeneratedClassTest.php [hunk 6]
```diff
@@ -269,6 +269,43 @@ public function testUint32Field()
...
+        $this->assertSame(MIN_INT32, $rf[2]);
+        $this->assertSame(-1, $rf[3]);
+    }
+
+    public function testMapUint32Uint32Boundaries()
+    {
+        $m = new TestMessage();
+        $map = $m->getMapUint32Uint32();
+
+        $map[1] = 0;
```
Comment (honglooker) [inline php/tests/GeneratedClassTest.php:302]: why not map[0] ?

**Review by honglooker — COMMENTED** [review]
Thanks, some q's below
